android/gltfio-android/src/main/java/com/google/android/filament/gltfio/FilamentAsset.java

```diff
@@ -99,6 +100,17 @@ public int popRenderables(@Nullable @Entity int[] entities) {
         return result;
     }
 
+    public @NonNull MaterialInstance[] getMaterialInstances() {
+        final int count = nGetMaterialInstanceCount(mNativeObject);
+        MaterialInstance[] result = new MaterialInstance[count];
+        long[] natives = new long[count];
+        nGetMaterialInstances(mNativeObject, natives);
+        for (int i = 0; i < count; i++) {
```

Comment:
Something you might want to double-check, when I tried hacking this myself for my demo I ran into a weird issue where some of the material instances where `nullptr`, which would lead to crashes in `->getMaterial()` later on. My fix was to first count the number of non nullptr instances and use that as the real count. I'll share the glTF files with you that were causing the crashes.

Reply:
I haven't been able to repro with that asset (on native anyway).